zerver/lib/streams.py

```diff
@@ -658,7 +658,7 @@ def filter_stream_authorization(
             continue
 
         # Members and administrators are authorized for public streams
-        if not stream.invite_only and not user_profile.is_guest:
+        if not stream.invite_only and not user_profile.is_guest and not stream.deactivated:
```

Comment:
Why is this security/authorization change in this commit? It's very concerning to see this squashed in with a commit about name changes, because that's a common way to accidentally introduce security bugs.

This reinforces my opinion in https://github.com/zulip/zulip/pull/28478/commits/0601fc653b6d87224869d95ad54c23d5ab3fa163#r1500104925 that you should remove the `invite_only` changes from this commit.

Reply:
I agree with you. We should leave it for a next PR.